File: src/main/java/me/andre111/d20server/scripting/variable/parser/VariableParserTokenList.java
package me.andre111.d20server.scripting.variable.parser;

import me.andre111.d20server.scripting.ParserContext;
import me.andre111.d20server.scripting.ScriptException;
import me.andre111.d20server.scripting.variable.TokenListVariable;
import me.andre111.d20server.scripting.variable.Variable;

public class VariableParserTokenList extends VariableParser {
	@Override
	public Variable parse(ParserContext context, String fullName, String name) throws ScriptException {
		if(context.getTokenFinder() == null) throw new ScriptException("Internal Error: No TokenFinder present!");
		
		return new TokenListVariable(fullName, name, context.getTokenFinder());
	}
}

File: core/common/scripting/variable/parser/variable-parser-token-list.js
import { VariableParser } from './variable-parser.js';
import { TokenListVariable } from '../token-list-variable.js';
import { EntityFinderList } from '../../finder/entity-finder-list.js';

export class VariableParserTokenList extends VariableParser {
    constructor() {
        super();
    }

    parse(context, fullName, name) {
        if(!context.getEntityFinder('token')) throw new Error(`Internal Error: No EntityFinder for token present`);

        return new TokenListVariable(fullName, false, context.getEntityFinder('token'), new EntityFinderList(name));
    }
}
